src/common/transformations/src/transformations/convert_precision.cpp

```diff
@@ -335,6 +338,20 @@ bool fuse_type_to_random_uniform_v8(const std::shared_ptr<ngraph::Node>& node, o
     return false;
 }
 
+bool fuse_type_to_unique_v10(const std::shared_ptr<Node>& node, ov::element::Type to, size_t idx) {
+    if (auto unique = ov::as_type_ptr<opset10::Unique>(node)) {
+        if (to == ov::element::i32 || to == ov::element::i64) {
+            if (idx == 1 || idx == 2) {
+                unique->set_index_element_type(to);
+            } else if (idx == 3) {
+                unique->set_count_element_type(to);
```

Comment:
it looks like we need to `return true` if we set a new type

Reply:
I thought that too, but the conversion is not called for each output when `fuse_to_...` function returns `true` and it looks like `true` value should be used when `replace_node` is used:
https://github.com/openvinotoolkit/openvino/blob/857ba7506ecadec2eacde6e842c725ad58aebbcd/src/common/transformations/src/transformations/convert_precision.cpp#L144-L146
Other way `return true` breaks the loop and the loop in `convert_node_output_precision` doesn't iterate over all of the outputs.
https://github.com/openvinotoolkit/openvino/blob/857ba7506ecadec2eacde6e842c725ad58aebbcd/src/common/transformations/src/transformations/convert_precision.cpp#L133

Another solution is to call `set_index_element_type` and `set_count_element_type` regardless the `idx` variable, so for all of the outputs at once. Then return `true` can be used.

Reply:
Discussed offline, the adjustment of `convert_node_output_precision` will be added in a separate PR.
cc @tomdol